mkosi/__init__.py

```diff
@@ -1693,71 +1693,79 @@ def clean_dnf_metadata(root: Path) -> None:
         *root.glob("var/log/hawkey.*"),
         root / "var/cache/dnf",
     ]
-    keep_dnf_data = os.access(root / "bin/dnf", os.F_OK, follow_symlinks=False)
 
-    if keep_dnf_data or not any(os.path.exists(path) for path in dnf_metadata_paths):
+    cond = always or os.access(root / "bin/dnf", os.F_OK, follow_symlinks=False)
+
+    if  not cond or not any(os.path.exists(path) for path in dnf_metadata_paths):
         return
 
     with complete_step("Cleaning dnf metadata…"):
         remove_glob(*dnf_metadata_paths)
 
 
-def clean_yum_metadata(root: Path) -> None:
-    """Removes yum metadata iff /bin/yum is not present in the image"""
+def clean_yum_metadata(root: Path, always: bool) -> None:
+    """Remove yum metadata if /bin/yum is not present in the image"""
     yum_metadata_paths = [
         root / "var/lib/yum",
         *root.glob("var/log/yum.*"),
         root / "var/cache/yum",
     ]
-    keep_yum_data = os.access(root / "bin/yum", os.F_OK, follow_symlinks=False)
 
-    if keep_yum_data or not any(os.path.exists(path) for path in yum_metadata_paths):
+    cond = always or os.access(root / "bin/yum", os.F_OK, follow_symlinks=False)
```

Comment:
@keszybz this changed the default to always remove metadata if the binary is present - was that intentional?